File: src/main/kotlin/xyz/mahmoudahmed/dsl/Platymap.java
package xyz.mahmoudahmed.dsl;

import xyz.mahmoudahmed.adapter.InputAdapterService;
import xyz.mahmoudahmed.dsl.Format;
import xyz.mahmoudahmed.dsl.FunctionBuilder;
import xyz.mahmoudahmed.dsl.SourceBuilder;
import xyz.mahmoudahmed.dsl.TypedSourceBuilder;
import xyz.mahmoudahmed.format.FormatType;


public class Platymap {
    private static final InputAdapterService adapterService = new InputAdapterService();

    // Entry point factory methods
    public static SourceBuilder flow(String source) {
        return new SourceBuilder(source);
    }

    public static <S> TypedSourceBuilder<S> flow(Class<S> sourceClass) {
        return new TypedSourceBuilder<>(sourceClass);
    }

    public static FunctionBuilder function(String name) {
        return new FunctionBuilder(name);
    }

    // Utility method to convert Format enum to Kotlin's FormatType
    static FormatType toKotlinFormat(Format format) {
        switch (format) {
            case JSON: return FormatType.JSON;
            case XML: return FormatType.XML;
            case CSV: return FormatType.CSV;
            case YAML: return FormatType.YAML;
            case PROPERTIES: return FormatType.PROPERTIES;
            default: return FormatType.UNKNOWN;
        }
    }


    // Get the adapter service instance
    public static InputAdapterService getAdapterService() {
        return adapterService;
    }
}

File: src/main/kotlin/xyz/mahmoudahmed/dsl/core/Platymap.kt
package xyz.mahmoudahmed.dsl.core

import xyz.mahmoudahmed.adapter.InputAdapterService
import xyz.mahmoudahmed.dsl.functions.FunctionBuilder
import xyz.mahmoudahmed.dsl.builders.SourceBuilder
import xyz.mahmoudahmed.dsl.typed.TypedSourceBuilder

/**
 * The main entry point for the mapping DSL.
 */
object Platymap {
    private val adapterService = InputAdapterService()

    /**
     * Starts a new mapping flow with the specified source name.
     *
     * @param source The name of the source
     * @return A builder for configuring the source
     */
    fun flow(source: String): SourceBuilder {
        return SourceBuilder(source)
    }

    /**
     * Starts a new typed mapping flow with the specified source class.
     *
     * @param sourceClass The class of the source
     * @return A builder for configuring the typed source
     */
    fun <S> flow(sourceClass: Class<S>): TypedSourceBuilder<S> {
        return TypedSourceBuilder(sourceClass)
    }

    /**
     * Creates a new function with the specified name.
     *
     * @param name The name of the function
     * @return A builder for configuring the function
     */
    fun function(name: String): FunctionBuilder {
        return FunctionBuilder(name)
    }

    /**
     * Gets the adapter service.
     *
     * @return The adapter service
     */
    fun getAdapterService(): InputAdapterService {
        return adapterService
    }
}
